packages/optimizer/lib/fetchRuntimeParameters.js

```diff
@@ -149,11 +155,17 @@ async function fetchAmpRuntimeVersion_(context) {
  * @private
  */
 async function fetchLatestRuntimeData_({config, ampUrlPrefix, lts}, versionKey = null) {
-  const ampRuntimeData = {
+  let ampRuntimeData;
+  ampRuntimeData = {
     version: await config.runtimeVersion.currentVersion({ampUrlPrefix, lts}),
     maxAge: MaxAge.create(AMP_RUNTIME_MAX_AGE).toObject(),
   };
-  if (versionKey) {
+  if (!ampRuntimeData.version && ampUrlPrefix !== AMP_CACHE_HOST) {
+    config.log.error(
+      `Could not download runtime version from ${ampUrlPrefix}. Falling back to ${AMP_CACHE_HOST}`
+    );
+    ampRuntimeData = await fetchLatestRuntimeData_({config, AMP_CACHE_HOST, lts}, versionKey);
+  } else if (versionKey) {
```

Comment:
Do you want to store bad results? If not:
`} else if (ampRuntimeData.version && versionKey) {`

Reply:
Done